File: Codechef/CHF_Heist/CHF_Heist.py
T = int(input())
while T:
    T -= 1
    D, d, P, Q = map(int, input().split())
    res = ((d * (D // d) * (2 * P + ((D // d) - 1) * Q)) // 2) + (
        (D % d) * (P + (D // d) * Q)
    )
    print(res)

File: Codechef/CHF_Heist/CHF_Heist.cpp
#include <iostream>
using namespace std;

int main()
{
    int T;
    cin >> T;
    while (T--)
    {
        long long D, d, P, Q;
        cin >> D >> d >> P >> Q;
        long res = ((d * (D / d) * (2 * P + ((D / d) - 1) * Q)) / 2) + ((D % d) * (P + (D / d) * Q));
        cout << res << endl;
    }
    return 0;
}